cmd/peer-rest-server.go

```diff
@@ -61,225 +61,179 @@ func (s *peerRESTServer) GetLocksHandler(w http.ResponseWriter, r *http.Request)
-	if !s.IsValid(w, r) {
-		s.writeErrorResponse(w, errors.New("Invalid request"))
-		return
-	}
-
+func (s *peerRESTServer) DeleteUserHandler(mss *grid.MSS) (np grid.NoPayload, nerr *grid.RemoteErr) {
 	objAPI := newObjectLayerFn()
 	if objAPI == nil {
-		s.writeErrorResponse(w, errServerNotInitialized)
-		return
+		return np, grid.NewRemoteErr(errServerNotInitialized)
 	}
 
-	vars := mux.Vars(r)
-	accessKey := vars[peerRESTUser]
+	accessKey := mss.Get(peerRESTUser)
 	if accessKey == "" {
-		s.writeErrorResponse(w, errors.New("username is missing"))
-		return
+		return np, grid.NewRemoteErr(errors.New("username is missing"))
 	}
 
-	if err := globalIAMSys.DeleteUser(r.Context(), accessKey, false); err != nil {
-		s.writeErrorResponse(w, err)
-		return
+	if err := globalIAMSys.DeleteUser(context.Background(), accessKey, false); err != nil {
+		return np, grid.NewRemoteErr(err)
 	}
+
+	return
 }
 
 // LoadUserHandler - reloads a user on the server.
-func (s *peerRESTServer) LoadUserHandler(w http.ResponseWriter, r *http.Request) {
-	if !s.IsValid(w, r) {
-		s.writeErrorResponse(w, errors.New("Invalid request"))
-		return
-	}
-
+func (s *peerRESTServer) LoadUserHandler(mss *grid.MSS) (np grid.NoPayload, nerr *grid.RemoteErr) {
 	objAPI := newObjectLayerFn()
 	if objAPI == nil {
-		s.writeErrorResponse(w, errServerNotInitialized)
-		return
+		return np, grid.NewRemoteErr(errServerNotInitialized)
 	}
 
-	vars := mux.Vars(r)
-	accessKey := vars[peerRESTUser]
+	accessKey := mss.Get(peerRESTUser)
 	if accessKey == "" {
-		s.writeErrorResponse(w, errors.New("username is missing"))
-		return
+		return np, grid.NewRemoteErr(errors.New("username is missing"))
 	}
 
-	temp, err := strconv.ParseBool(vars[peerRESTUserTemp])
+	temp, err := strconv.ParseBool(mss.Get(peerRESTUserTemp))
 	if err != nil {
-		s.writeErrorResponse(w, err)
-		return
+		return np, grid.NewRemoteErr(err)
 	}
 
 	userType := regUser
 	if temp {
 		userType = stsUser
 	}
 
-	if err = globalIAMSys.LoadUser(r.Context(), objAPI, accessKey, userType); err != nil {
-		s.writeErrorResponse(w, err)
-		return
+	if err = globalIAMSys.LoadUser(context.Background(), objAPI, accessKey, userType); err != nil {
+		return np, grid.NewRemoteErr(err)
 	}
+
+	return
 }
 
 // LoadGroupHandler - reloads group along with members list.
-func (s *peerRESTServer) LoadGroupHandler(w http.ResponseWriter, r *http.Request) {
-	if !s.IsValid(w, r) {
-		s.writeErrorResponse(w, errors.New("Invalid request"))
-		return
-	}
-
+func (s *peerRESTServer) LoadGroupHandler(mss *grid.MSS) (np grid.NoPayload, nerr *grid.RemoteErr) {
 	objAPI := newObjectLayerFn()
 	if objAPI == nil {
-		s.writeErrorResponse(w, errServerNotInitialized)
-		return
+		return np, grid.NewRemoteErr(errServerNotInitialized)
 	}
 
-	vars := mux.Vars(r)
-	group := vars[peerRESTGroup]
-	err := globalIAMSys.LoadGroup(r.Context(), objAPI, group)
+	group := mss.Get(peerRESTGroup)
```

Comment:
Maybe check here is group is not missing? (We are doing this for the other handlers)

Reply:
I will get to these subsequently @fwessels. Thanks in another PR.